tests/fixtures/when.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fixtures
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	git "github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/numaproj-labs/numaplane/pkg/apis/numaplane/v1alpha1"
+	planepkg "github.com/numaproj-labs/numaplane/pkg/client/clientset/versioned/typed/numaplane/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var auth = &http.BasicAuth{
+	Username: "root",
+	Password: "root",
+}
+
+type When struct {
+	t             *testing.T
+	restConfig    *rest.Config
+	kubeClient    kubernetes.Interface
+	gitSync       *v1alpha1.GitSync
+	gitSyncClient planepkg.GitSyncInterface
+}
+
+func (w *When) CreateGitSyncAndWait() *When {
+
+	w.t.Helper()
+	if w.gitSync == nil {
+		w.t.Fatal("No GitSync to create")
+	}
+	w.t.Log("Creating GitSync", w.gitSync.Name)
+	ctx := context.Background()
+	i, err := w.gitSyncClient.Create(ctx, w.gitSync, metav1.CreateOptions{})
+	if err != nil {
+		w.t.Fatal(err)
+	} else {
+		w.gitSync = i
+	}
+	// wait for GitSync to run
+	if err := w.waitForGitSyncRunning(ctx, w.gitSyncClient, w.gitSync.Name, defaultTimeout); err != nil {
+		w.t.Fatal(err)
+	}
+	return w
+
+}
+func (w *When) DeleteGitSyncAndWait() *When {
+
+	w.t.Helper()
+	if w.gitSync == nil {
+		w.t.Fatal("No gitsync to delete")
+	}
+	w.t.Log("Deleting gitsync", w.gitSync.Name)
+	ctx := context.Background()
+	if err := w.gitSyncClient.Delete(ctx, w.gitSync.Name, metav1.DeleteOptions{}); err != nil {
+		w.t.Fatal(err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
```

Comment:
Since you have `waitForGitSyncRunning()` above, maybe we should have `waitForGitSyncDeleted()` for symmetry?